### PR TITLE
Adds `onCall` function support (#9)

### PR DESCRIPTION
**Pull request by abehaskins**
### Description

Adds support for `https.onCall` functions and improves filtering of `https.onRequest` functions.

**Do not merge!** I'm not sure if there are other aspects of this I'm missing. I'm not sure if there's any value to adding `providers/https.ts` other than to allow `onCall` support to be discoverable in the library. Since the bundles passed to the functions are arbitrary JSON I'm not sure what help we could be.

Let me know your thoughts!

### Code sample

See [providers/https.spec.ts](https://github.com/firebase/firebase-functions-test/blob/ah-oncall/spec/providers/https.spec.ts)

### CODE REVIEW

**Comment by abehaskins** [comment]
Is there somewhere I can see what `CallableContext` should be?

**Comment by abehaskins** [comment]
JK, [found it](https://github.com/firebase/firebase-functions/blob/28dd25c42f63b2baed82f7449d0729e859efbbf4/src/providers/https.ts#L253) I'll work on adding support for this context type and moving to lodash.

**Comment by rilian** [comment]
thank you for working on this!

File: spec/main.spec.ts [hunk 1]
```diff
@@ -88,6 +88,15 @@ describe('main', () => {
       expect(context.authType).to.equal('USER');
     });
 
+    it('should allow allow invalid options', () => {
```
Comment (laurenzlong) [inline spec/main.spec.ts]: should not allow

File: spec/main.spec.ts [hunk 2]
```diff
@@ -88,6 +88,15 @@ describe('main', () => {
...
+      const wrapped = wrap(constructCF());
+      expect(() => wrapped('data', {
+          auth: { uid: 'abc' },
+          authType: 'USER',
```
Comment (laurenzlong) [inline spec/main.spec.ts]: remove line 95, since authType is also invalid for callables

File: src/main.ts [hunk 3]
```diff
@@ -34,53 +34,97 @@ export type EventContextOptions = {
    * If omitted, random values will be generated.
    */
   params?: { [option: string]: any };
-  /** (Only for database functions.) Firebase auth variable representing the user that triggered
+  /** (Only for database functions and https.onCall.) Firebase auth variable representing the user that triggered
    *  the function. Defaults to null.
    */
   auth?: any;
-  /** (Only for database functions.) The authentication state of the user that triggered the function.
+  /** (Only for database and functions.) The authentication state of the user that triggered the function.
```
Comment (laurenzlong) [inline src/main.ts]: and https.onCall functions

**Pull request merged**